Prolog/models/terminal.pl

```diff
@@ -0,0 +1,89 @@
+:- module(terminal,[getTerminalWidth/1,getTerminalHeight/1,startPlayerInputThread/0,fetchFromThread/1]).
+
+getTerminalWidth(Width):- tty_size(_, Width).
+getTerminalHeight(Height):- tty_size(Height, _).
+
+startPlayerInputThread:- 
+thread_create(getPlayerInput, Id).
+
+getPlayerInput:- 
+  get_key(NewInput),
+  handlePlayerInput(NewInput),
+  getPlayerInput.
+
+get_key(X):-                                         

```

Comment:
Are we going to follow `camelCase`? If so, what do you think about renaming this rule to `getKey` for consistency?